### ISSUE TITLE
Nautilus-dropbox suddenly stuck ("Starting...") after reboot on four different machines

### ISSUE THREAD

**Issue by dushoff** [dropbox/nautilus-dropbox#149]
I have four different Ubuntu 24.04 machines that have all been running nautilus-dropbox successfully for months to years, some since Ubuntu 20.04. Starting some time last week, they all successively stopped syncing after reboot. I am not aware of anything that I changed (and it doesn't seem likely that I changed anything on all four machines).

My dropbox status, even after hours, now always says:

```
🌑 dushoff@siX:~/screens$ dropbox status
/usr/bin/dropbox:613: DeprecationWarning: isSet() is deprecated, use is_set() instead
  if self.stop_event.isSet(): break
Starting...
🌑
```

`dropbox update` succeeds but does not change the situation. 

I get occasionaly warnings about fs.inotify.max_user_watches, but this seems to be a red herring. Warnings don't always stop even when I increase this value well beyond my folder number, and behaviour doesn't change even when the warnings stop.

Here is some more information:

```
🌑 $  lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.1 LTS
Release:        24.04
Codename:       noble
```

```
🌑 $ dpkg -l nautilus-dropbox
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name             Version              Architecture Description
+++-================-====================-============-========================>
ii  nautilus-dropbox 2019.02.14-1.2build2 amd64        Dropbox integration for >
```

I'm not using firewalls or vpn. The machines are in two different physical locations (two at home and two at work).

I tried the dropbox "headless" approach on one machine. This installs but produces the same forever "Starting..." behaviour. I tried other things, but this one seems potentially relevant to the nautilus-dropbox problem.

**Comment by yioannides** [comment]
I had this exact problem (for a most likely different reason) about a year ago, but one thing I was told by Dropbox's customer service is that in order for Dropbox to work properly, there is a hidden cap of 300,000 files, as well as the inability to sync aliased files & folders with weird permissions, these two things may mess with Dropbox's boot sequence. I hope this helps a bit.

> ii  nautilus-dropbox 2019.02.14-1.2build2 amd64

Also, unless you're running on an LTS system, updating Dropbox might also help...

**Comment by dushoff** [comment]
Thanks. I tried all of the updates, and -- mostly based on my experience with Dropbox "support" -- am trying hard to just drop Dropbox.

**Comment by yioannides** [comment]
I am personally very satisfied with the Dropbox customer support (I have a Dropbox Plus account, so maybe that helps), but the software itself is really not optimized for Linux and GNOME in specific.

**Comment by dushoff** [comment]
I'm glad you had a better experience. I am paying a fair amount of money, but got kept on the chat line with what must have been a human-supervised bot for over an hour, and never got satisfactory followup. I was given the same set of instructions about software and configuration many times (on both chat and forum); there was no sign that anyone ever recognized that it must be something about my files (since it stopped working on four machines around the same time).

**Comment by yioannides** [comment]
Yeah, I'm sorry to hear that man. It is possible they switched to a human-supervised bot in the last year. However, it wouldn't hurt checking the tips I mentioned above, because it could be something similar. Also, deleting `~/.dropbox` every now and then has also helped me with some other bugs.

**Comment by cversek** [comment]
Ran into this exact behavior (dropbox status stuck on "Starting..." forever, daemon 262.4.3183, Ubuntu, ext4) and got it fixed. It turned out to be two independent problems that mask each other:

1. The daemon's embedded browser hangs during display init on a Wayland session with an NVIDIA GPU. It never finishes starting, so it never gets to login/index/sync. The tell in stderr: repeated `pci id ... driver (null)` GPU-probe lines and GDK-Wayland assertions. `GDK_BACKEND=x11` quiets some of it but doesn't fix the hang. This is also why the "headless" install hangs - the daemon still tries to bring up its embedded UI.

2. The inotify warning is not entirely a red herring, but the usual advice undershoots: the limit has to exceed your *directory* count, not just feel big. Mine was 100000 against a 125k-directory tree - warnings persisted and watching was incomplete. Count them with `find "$HOME/Dropbox" -type d | wc -l`, then size `fs.inotify.max_user_watches` above that (I used 524288).

The fix that made it permanent for me: run the real daemon under a virtual X display so it never touches the GPU or Wayland at all.

```
sudo apt install xvfb
# quick test:
xvfb-run -a ~/.dropbox-dist/dropboxd &
dropbox status   # went from "Starting..." to "Indexing..." immediately
```

Made permanent with a systemd user unit (`ExecStart=/usr/bin/xvfb-run -a %h/.dropbox-dist/dropboxd`, `WantedBy=default.target`) plus `sudo loginctl enable-linger $USER` so it starts at boot without a login. Sync came back instantly after months frozen, including a ~1M-file backlog.

If your stderr shows the GPU-probe/GDK spam when launching `dropboxd` from a terminal, this is probably your problem too.

For transparency: I tracked this down and worked out the fix using Claude Fable 5 running in Claude Code.

---

**Update, a day on** — three corrections after living with the fix, one of them to my own advice above.

**1. Don't put the sysctl in `/etc/sysctl.conf`; it isn't durable.** Mine silently reverted to 65536 and the "Unable to monitor entire Dropbox folder hierarchy" warning came back. `/usr/lib/sysctl.d/30-localsearch.conf` ships `fs.inotify.max_user_watches = 65536` and wins over `sysctl.conf`. Put yours in `/etc/sysctl.d/99-<name>.conf` instead: `/etc/sysctl.d` overrides `/usr/lib/sysctl.d`, and the `99-` prefix sorts last so it also survives any future vendor drop-in whatever number it carries. Check the result with `sysctl -n fs.inotify.max_user_watches`, not by re-reading the file you just wrote.

**2. Careful with my quick test above — `xvfb-run ... &` can freeze instead of starting.** `xvfb-run` shells out to `stty size`. Backgrounded from a terminal it sits outside the foreground process group, so touching the TTY earns it `SIGTTOU` and the whole tree stops: `ps` shows `STAT=T`, and `dropboxd` never spawns at all. It looks exactly like a hung daemon, but the daemon was never started — the only processes are shells. `SIGTERM` won't clear it either, because a stopped process defers catchable signals; it needs `SIGKILL`. Redirect stdin if you background it (`xvfb-run -a ~/.dropbox-dist/dropboxd </dev/null &`), or just test through the unit. This doesn't affect the systemd unit, where there's no controlling TTY and `stty` fails harmlessly.

**3. `Restart=on-failure` is the wrong policy here.** When the daemon gives up on display init it exits with status **0**, which `on-failure` reads as success — so it stays dead and nothing is logged as an error. `Restart=always` recovers it. If you add that, put `StartLimitIntervalSec`/`StartLimitBurst` in `[Unit]`, not `[Service]` — systemd ignores them in `[Service]` with only a warning, so the limit looks configured while doing nothing.

Also worth adding `Environment=GDK_BACKEND=x11` to the unit alongside `xvfb-run`. On its own it doesn't fix the hang, as I said above — but a systemd *user* unit inherits the session environment including `WAYLAND_DISPLAY`, and GTK will then prefer Wayland over the X display `xvfb-run` just created, so Xvfb runs and gets ignored. Pinning the backend forces it onto the virtual display. The tell that it's actually working: the GDK-Wayland assertions stop appearing entirely, rather than just thinning out.

Otherwise still holding up. My directory count had grown to ~134k since; 524288 still leaves ample headroom.

For transparency: this update was worked out with Claude Opus 5 in Claude Code.

**Comment by dushoff** [comment]
Glad it worked for you. Dropbox help was so dishonest with me that I just cancelled my account and I've been more or less suffering ever since, but I won't go back. If it had happened in the LLM era I might have had a better experience.
